bin/Main.purs

```diff
@@ -422,3 +472,22 @@ tokenStreamToArray = go []
       Left err
     TokenCons tok _ next _ ->
       go (Array.snoc acc tok.value) next
+
+partitionCheckedFiles
+  :: Array WorkerOutput
+  -> { errors :: Array (Tuple FilePath String), notFormatted :: Array FilePath }
+partitionCheckedFiles = do
+  let
+    foldFn { errors, notFormatted } result = do
+      let
+        errors'
+          | String.null result.error = errors
+          | otherwise = Array.cons (Tuple result.filePath result.error) errors
+
+        notFormatted'
+          | result.alreadyFormatted = notFormatted
+          | otherwise = Array.cons result.filePath notFormatted
+
+      { errors: errors', notFormatted: notFormatted' }
+
+  foldl foldFn mempty
```

Comment:
This could be written as a `foldMap` and then you wouldn't need the `cons` and stuff.
```purescript
foldMap \result ->
  { errors: guard (not (String.null result.error)) [ ... ]
  , notFormatted: guard (not result.alreadyFormatted) [ ... ]
  }
```

Reply:
That's much nicer. Thanks!